checkout/payment_flow.py

```
@@ -0,0 +1,63 @@
+class CheckoutService:
+    """Loads an order, validates it, then initiates payment."""
+
+    def __init__(self, orders, payments, db):
+        self.orders = orders
+        self.payments = payments
+        self.db = db
+
+    def load_order(self, order_id):
+        order = self.orders.fetch_order(order_id)
+        row = self.db.execute(
+            "SELECT status FROM orders WHERE id = '" + str(order_id) + "'"
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)